src/buildx/history.ts

```diff
@@ -85,6 +101,72 @@ export class History {
     core.info(`exporting build record to ${outDir}`);
     fs.mkdirSync(outDir, {recursive: true});
 
+    if (opts.useContainer || (await this.buildx.versionSatisfies('<0.23.0'))) {
+      return await this.exportLegacy(builderName, nodeName, refs, outDir, opts.image);
+    }
+
+    // wait 3 seconds to ensure build records are finalized: https://github.com/moby/buildkit/pull/5109
+    await Util.sleep(3);
```

Comment:
Ah, I forgot this. Buildx finalizes trace on `buildx history trace` automatically. This could be used in here instead but would be messy. We should update `buildx history export` to do it as well (eg. when a flag is set).

Reply:
> We should update `buildx history export` to do it as well (eg. when a flag is set).

Yes sounds good

Reply:
Opened https://github.com/docker/buildx/issues/3149